src/main/java/seedu/address/model/person/Person.java

```diff
@@ -24,78 +26,207 @@ public class Person {
-    private Set<Tag> tags;
+    private final Address address;
+    private final Set<Tag> tags;
     private final ObservableList<Person> pairedPersons;
     private final ObservableList<Person> unmodifiablePairedPersons;
     private final List<Person> unmodifiablePairingsView;
 
+    private final PersonBuilder personBuilder;
+
     /**
      * The Builder for the Person class.
      */
-    public static class PersonBuild<T extends PersonBuild<T>> {
+    public static class PersonBuilder extends Builder<PersonBuilder> {
         // Required parameters
-        private final Name name;
+        private Name name;
 
         // Optional parameters - initialized to default values
-        private Phone phone = new Phone("000");
-        private Email email = new Email("default@email");
-        private Address address = new Address("Default Address");
-        private Set<Tag> tags = new HashSet<>();
-        private ObservableList<Person> pairedPersons = FXCollections.observableArrayList();
+        private Phone phone;
+        private Email email;
+        private Address address;
+        private Set<Tag> tags;
+        private ObservableList<Person> pairedPersons;
+
+        /**
+         * Constructor for PersonBuilder.
+         */
+        public PersonBuilder() {
+
+        }
+
+        /**
+         * Constructor for PersonBuilder with Person.
+         */
+        public PersonBuilder(Person p) {
+            this.name = p.getName();
+            this.phone = p.getPhone();
+            this.email = p.getEmail();
+            this.address = p.getAddress();
+            this.tags = p.getTags();
+            this.pairedPersons = FXCollections.observableArrayList();
+            this.pairedPersons.setAll(p.getPairings());
+        }
+
+        /**
+         * Copy constructor.
+         */
+        public PersonBuilder(PersonBuilder toCopy) {
+            this.name = toCopy.name;
+            this.phone = toCopy.phone;
+            this.email = toCopy.email;
+            this.address = toCopy.address;
+            this.tags = toCopy.tags;
+            this.pairedPersons = toCopy.pairedPersons;
+        }
+
+        /**
+         * Setter for the name parameter.
+         */
+        public PersonBuilder name(Name name) {
+            if (name != null) {
+                this.name = name;
+            }
+            return this;
+        }
 
         /**
-         * Constructor for PersonBuild.
+         * Setter for the name parameter but with String argument.
          */
-        public PersonBuild(Name name) {
-            requireAllNonNull(name);
-            this.name = name;
+        public PersonBuilder name(String name) {
+            return this.name(new Name(name));
+        }
+
+        /**
+         * Setter for the name parameter but only if this.name does not already exist.
+         */
+        public PersonBuilder nameIfNotPresent(Name name) {
+            if (this.name == null && name != null) {
```

Comment:
Nit: Is the `name` parameter (not `this.name`) ever supposed to take the value of `null`? I tried tracing code for `edit` command with the debugger and I don't think this parameter ever takes the value of `null`. Should we throw a `NullPointerException` if it is `null`?

```suggestion
            requireNonNull(name);
            if (this.name == null) {
```

Same for other `xyzIsNotPresent` methods.